### PR TITLE
Detect directories to notify incorrect use of Import API

### PR DESCRIPTION
**Pull request by iriberri**
* [x] Before merging, update docs accordingly: http://docs.cartodb.com/cartodb-editor/error-codes-and-solutions/

Fixes https://github.com/CartoDB/cartodb/issues/6307

### CODE REVIEW

**Comment by iriberri** [comment]
The details to reproduce the issue can be found: https://github.com/CartoDB/cartodb/issues/6307#issuecomment-173221350

**Comment by iriberri** [comment]
Data Import result from reproduction in staging:

```json
{
id: "591109a2-62fb-4fde-b973-f4ee3b70a50a",
user_id: "868aad2f-e085-441a-afb8-7adb7c0802c6",
table_id: null,
data_type: "datasource",
table_name: null,
state: "failure",
error_code: 1010,
queue_id: "591109a2-62fb-4fde-b973-f4ee3b70a50a",
tables_created_count: null,
synchronization_id: null,
type_guessing: true,
quoted_fields_guessing: true,
content_guessing: false,
create_visualization: false,
visualization_id: null,
user_defined_limits: "{"twitter_credits_limit":0}",
get_error_text: null,
display_name: "591109a2-62fb-4fde-b973-f4ee3b70a50a",
success: false,
original_url: "",
warnings: null,
is_raster: false
}
```

**Comment by iriberri** [comment]
Please, @rafatower, take a look when you can. Thanks! :-)

File: services/importer/lib/importer/unp.rb [hunk 1]
```diff
@@ -46,6 +46,8 @@ def run(path)
       end
 
       def without_unpacking(path)
+        raise NotAFileError if File.directory?(path)
```
Comment (rafatower) [inline services/importer/lib/importer/unp.rb]: better use
```ruby
raise NotAFileError if File.file?(path)
```
http://ruby-doc.org/core-2.2.3/File.html#method-c-file-3F

if that's precisely what you mean to do

File: services/importer/spec/unit/unp_spec.rb [hunk 2]
```diff
@@ -41,7 +41,11 @@
       unp.without_unpacking(zipfile_factory)
       unp.source_files.size.should eq 1
     end
-  end #withount_unpacking
+
+    it 'raises if the path does not belong to a file' do
+      expect { Unp.new.without_unpacking('/var/tmp') }.to raise_error NotAFileError
+    end
+  end #without_unpacking
```
Comment (rafatower) [inline services/importer/spec/unit/unp_spec.rb]: I'd take advantage of this CR to remove the `#without_unpacking` line

**Comment by rafatower** [comment]
basically ok, consider my comments (minor things)

File: services/importer/lib/importer/unp.rb (hunk 1 above)
Comment (iriberri) [inline services/importer/lib/importer/unp.rb]: Hm, not sure if I'm following.

Your line would raise an exception for any valid file, right? I want to raise an exception in the case that what is sent _is not_ a file (which could happen through the Import API).

I guess that a:

````
raise NotAFileError if !File.file?(path)
`````

could work too!

File: services/importer/spec/unit/unp_spec.rb (hunk 2 above)
Comment (iriberri) [inline services/importer/spec/unit/unp_spec.rb]: Always a pleasure! :-)

File: services/importer/lib/importer/unp.rb (hunk 1 above)
Comment (rafatower) [inline services/importer/lib/importer/unp.rb]: correct, sorry my bad, BIG typo :dancer:

**Comment by iriberri** [comment]
Just proposed a doc change [here](https://github.com/CartoDB/docs/pull/652). As soon as it's accepted this PR is ready to be deployed.

**Comment by rafatower** [comment]
:+1:

**Pull request merged**